Mol_Analysis.py

```diff
@@ -368,6 +368,50 @@ def rmsd(self, **kwargs):
                 print("Calculating RMSD")
                 excode, out, err = g_rmsd.run()
 
+    def rmsd_matrix(self, mainchain_group=5, rmsd_group=1, **kwargs):
+        """
+        Runs gmx rms and extracts the average RMSD.
+        """
+        rmsd_xvg = self.odir + "rmsd.xvg"
+        rmsd_file = self.odir + "rmsd_matrix.xpm" 
+
+        g_rms = gromacs.tools.G_rms(
+            f=self.outf,
+            s=self.outp,
+            o=rmsd_xvg,
+            m=rmsd_file,
+            input=(str(mainchain_group), str(rmsd_group))
+        )
+
+        if not self.dryrun:
+            print("Calculating RMSD matrix and extracting average RMSD...")
+            excode, out, err = g_rms.run()
+            if excode != 0:
+                print("gmx rms failed for RMSD matrix calculation.")
+                return None
+
+            avg_rmsd = None
+            lines = out.split('\n') + err.split('\n')
+            for line in lines:
+                if "RMSD:" in line and "Avg" in line:
+                    parts = line.strip().split(',')
+                    for p in parts:
+                        p = p.strip()
+                        if p.startswith("Avg"):
+                            avg_str = p.split()[1]
+                            avg_rmsd = float(avg_str)
+                            break
```

Comment:
it would be more robust and reader-friendly to just use a simple regular expression, e.g. (untested code):
```
compiled_re = re.compile("^RMSD: Min .*, Max .*, Avg (.*)")
for line in lines:
    match = compiled_re.match(line)
    if match:
            avg_rmsd = match.groups()[0]
            # would be also good to check if len(groups) == 1
            print(f"Average RMSD: {avg_rmsd:.6f} nm")

            stats_file = os.path.join(self.odir, "rmsd_matrix_avg.txt")
            np.savetxt(stats_file, [avg_rmsd], header="RMSD Matrix Average (nm)", comments="", fmt="%.6f")
            return avg_rmsd
print("Could not find Avg RMSD in the output.")
return None
```